### DISCUSSION TITLE
problems with the versions of omada software controller / helper script and TPLINK Version

### DISCUSSION THREAD

**Discussion by bahaman1971** [community-scripts/ProxmoxVE#1113] [Q&A]
I know it's between the holidays and I may now be banished to the eternal sub-vaults of github because I don't quite understand the whole system yet (where do I post what) but the following problem currently exists: the current version of the script 
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/omada.sh)" from 
https://community-scripts.github.io/ProxmoxVE/scripts?id=omada results in version 5.14.32.4 (release date 11-15-2024) of the controller being installed. and the "update" function also installs this version as proxmox lxc. However, at https://support.omadanetworks.com/en/download/software/omada-controller/, 
version 5.15.6.7 is shown as a stable release from 12-05-2024. So thats an issue that should be fixed ..

As a note: if you follow the description for updating the software controller at https://www.omadanetworks.com/de/support/faq/4099/ after the application/installation via the proxmox helper script and then run the "update" function of the helper script in the LXC again, this leads to the software controller performing a "downgrade" and then (at least for me) freezing when restarting the controller (and also the LXC), or the start process does not lead to a usable start of the controller. if you have previously made a backup of the settings with the last version of the software controller, this can no longer be integrated into the software controller that is installed via the proxmox helper script (since it comes from a newer version than the one currently installed) and thus you lose all settings ... this is just a NOTE if you want to avoid trouble between the new years NEVER TOUCH A RUNNING SYSTEM :-) 
... HAPPY NEW YEAR TO EVERYONE

**Comment by derlbear** [comment]
Same here! The update fails because the script doesn't download the most recent version of Omada. 
In the script, this part fetches the link to the "new" version:
`latest_url=$(curl -s "https://support.omadanetworks.com/en/product/omada-software-controller/?resourceType=download" | \
  grep -o 'https://static\.tp-link\.com/upload/software/[^"]*linux_x64[^"]*\.deb' | \
  head -n 1)
  latest_version=$(basename "$latest_url")`
the result:
```
https://static.tp-link.com/upload/software/2024/202407/20240710/Omada_SDN_Controller_v5.14.26.1_linux_x64.deb
https://static.tp-link.com/upload/software/2024/202402/20240227/Omada_SDN_Controller_v5.13.30.8_linux_x64.deb
https://static.tp-link.com/upload/software/2024/202407/20240710/Omada_SDN_Controller_v5.14.26.1_linux_x64.deb
https://static.tp-link.com/upload/software/2024/202402/20240227/Omada_SDN_Controller_v5.13.30.8_linux_x64.deb
```
But the most recent version is
`https://static.tp-link.com/upload/software/2024/202412/20241224/Omada_SDN_Controller_v5.15.6.7_linux_x64.deb`

I don't know how to fix this, but the script should not be used for an update.

Edit: Maybe the curl URL should look like this:
`https://support.omadanetworks.com/en/download/software/omada-controller/`

**Comment by Kaimane** [comment]
If you execute curl on the URL `https://support.omadanetworks.com/en/product/omada-software-controller/?resourceType=download`, you will only get the .tar.gz variant of the latest omada version. Presumably the URL is adapted during runtime.
To programmatically determine the latest .deb version, use a different URL in script: `https://www.vigi.com/de/support/download/omada-software-controller/`
You can retrieve this URL via curl and receive the correct link to the latest .deb version.

`latest_url=$(curl -s "https://www.vigi.com/de/support/download/omada-software-controller/" | grep -o 'https://static\.tp-link\.com/upload/software/[^"]*linux_x64[^"]*\.deb' | head -n 1)
latest_version=$(basename "$latest_url")`

**Reply by DrAdrianJones** [reply]
how do I implement this with the script provided?

**Reply by Kaimane** [reply]
All you have to do is open the command line of your Omada LXC and execute the `update` command. The script will take care of updating your Omada installation to the latest version.

**Reply by DrAdrianJones** [reply]
Sorry, a bit of a noob question - what's the default username and password for the omada LXC on initial install?
Thanks

**Reply by Kaimane** [reply]
I have never installed the Omada controller as LXC, but as hardware (OC200). As far as I can remember, a setup wizard opens after the first start in which the credentials can be specified. So there is no default user and password.
